File: app/src/main/java/net/tlalka/android/fiszki/core/components/ActivityComponent.java
package net.tlalka.android.fiszki.core.components;

import dagger.Subcomponent;
import net.tlalka.android.fiszki.core.annotations.ActivityScope;
import net.tlalka.android.fiszki.core.modules.ActivityModule;
import net.tlalka.android.fiszki.view.activities.HelpActivity;
import net.tlalka.android.fiszki.view.activities.LessonActivity;
import net.tlalka.android.fiszki.view.activities.LessonListActivity;
import net.tlalka.android.fiszki.view.activities.LessonStatsActivity;
import net.tlalka.android.fiszki.view.activities.MainActivity;
import net.tlalka.android.fiszki.view.activities.SettingsActivity;
import net.tlalka.android.fiszki.view.activities.TestActivity;
import net.tlalka.android.fiszki.view.activities.TestListActivity;
import net.tlalka.android.fiszki.view.activities.TestStatsActivity;
import net.tlalka.android.fiszki.view.activities.WelcomeActivity;
import net.tlalka.android.fiszki.view.activities.WordsActivity;

@ActivityScope
@Subcomponent(modules = ActivityModule.class)
public interface ActivityComponent {

    void inject(MainActivity mainActivity);

    void inject(WelcomeActivity welcomeActivity);

    void inject(LessonActivity lessonActivity);

    void inject(LessonListActivity lessonListActivity);

    void inject(LessonStatsActivity lessonStatsActivity);

    void inject(TestActivity testActivity);

    void inject(TestListActivity testListActivity);

    void inject(TestStatsActivity testStatsActivity);

    void inject(WordsActivity wordsActivity);

    void inject(HelpActivity helpActivity);

    void inject(SettingsActivity settingsActivity);
}

File: app/src/main/java/net/tlalka/android/fiszki/core/components/ActivityComponent.kt
package net.tlalka.android.fiszki.core.components

import dagger.Subcomponent
import net.tlalka.android.fiszki.core.annotations.ActivityScope
import net.tlalka.android.fiszki.core.modules.ActivityModule
import net.tlalka.android.fiszki.view.activities.HelpActivity
import net.tlalka.android.fiszki.view.activities.LessonActivity
import net.tlalka.android.fiszki.view.activities.LessonListActivity
import net.tlalka.android.fiszki.view.activities.LessonScoreActivity
import net.tlalka.android.fiszki.view.activities.MainActivity
import net.tlalka.android.fiszki.view.activities.SettingsActivity
import net.tlalka.android.fiszki.view.activities.TestActivity
import net.tlalka.android.fiszki.view.activities.TestListActivity
import net.tlalka.android.fiszki.view.activities.TestScoreActivity
import net.tlalka.android.fiszki.view.activities.WelcomeActivity
import net.tlalka.android.fiszki.view.activities.WordsActivity

@ActivityScope
@Subcomponent(modules = [ActivityModule::class])
interface ActivityComponent {

    fun inject(mainActivity: MainActivity)

    fun inject(welcomeActivity: WelcomeActivity)

    fun inject(lessonActivity: LessonActivity)

    fun inject(lessonListActivity: LessonListActivity)

    fun inject(lessonScoreActivity: LessonScoreActivity)

    fun inject(testActivity: TestActivity)

    fun inject(testListActivity: TestListActivity)

    fun inject(testScoreActivity: TestScoreActivity)

    fun inject(wordsActivity: WordsActivity)

    fun inject(helpActivity: HelpActivity)

    fun inject(settingsActivity: SettingsActivity)
}
